tests/integration/questionnaire/test_questionnaire_csrf.py

```diff
@@ -8,38 +10,97 @@ class TestQuestionnaireInterstitial(IntegrationTestCase):
 
     def test_form_not_processed_with_no_csrf_token(self):
         token = create_token('interstitial_page', 'test')
-        resp = self.client.get('/session?token=' + token.decode(), follow_redirects=True)
+        self.client.get('/session?token=' + token.decode(), follow_redirects=True)
 
         post_data = {
-            'action[start_questionnaire]': 'Start survey'
         }
 
         resp = self.client.post(self.BASE_URL + 'favourite-foods/0/introduction', post_data)
 
-        self.assertEquals(200, resp.status_code)
+        self.assertEquals(401, resp.status_code)
 
     def test_form_not_processed_with_incorrect_csrf_token(self):
         token = create_token('interstitial_page', 'test')
-        resp = self.client.get('/session?token=' + token.decode(), follow_redirects=True)
+        self.client.get('/session?token=' + token.decode(), follow_redirects=True)
 
         post_data = {
-            'csrf_token': 'made-up-token',
-            'action[start_questionnaire]': 'Start survey'
+            'csrf_token': 'made-up-token'
         }
 
         resp = self.client.post(self.BASE_URL + 'favourite-foods/0/introduction', post_data)
 
-        self.assertEquals(200, resp.status_code)
+        self.assertEquals(401, resp.status_code)
 
     def test_form_processed_with_correct_csrf_token(self):
         token = create_token('interstitial_page', 'test')
-        resp = self.client.get('/session?token=' + token.decode(), follow_redirects=True)
+        self.client.get('/session?token=' + token.decode(), follow_redirects=True)
 
         post_data = {
-            'action[start_questionnaire]': 'Start survey'
         }
 
         resp = self.get_and_post_with_csrf_token(self.BASE_URL + 'favourite-foods/0/introduction', post_data)
 
         self.assertEquals(302, resp.status_code)
 
+    def test_given_valid_answers_when_posting_invalid_csrf_token_then_answers_not_saved(self):
+        # Given
```

Comment:
Not that there's anything to be done unless the integration test PR merges; I'll re-base and update this when it merges to the new format.